oar/core/ldap.py

```diff
@@ -0,0 +1,151 @@
+import logging
+
+from ldap3 import Server, Connection, ALL, SUBTREE
+from ldap3.core.exceptions import LDAPException
+from ldap3.utils.dn import parse_dn
+
+from oar.core.exceptions import LdapHelperException
+from oar.core.util import is_valid_email
+
+logger = logging.getLogger(__name__)
+
+
+class LdapHelper:
+    """
+    Helper class is used to acquire data from LDAP server.
+    """
+
+    MANAGER = "manager"
+    PRIMARY_MAIL = "rhatPrimaryMail"
+    UID = "uid"
+
+    def __init__(self):
+        self.server = Server("ldaps://ldap.corp.redhat.com", get_info=ALL)
+
+    def get_manager_email(self, user_email: str) -> str:
+        """
+        Get manager email for specified user email
+
+        Args:
+            user_email (str): User primary email
+
+        Returns:
+            str: Found manager email for specified user email
+        """
+
+        if user_email is None or not is_valid_email(user_email):
+            raise LdapHelperException(f"Specified email is not valid: {user_email}")
+
+        manager_id = self._get_manager_id(user_email)
```

Comment:
is it possible that `manager_id` is `None` here?

Reply:
I added one more check to `_get_manager_id` and move the check from `get_manager_email` to `_get_manager_id`

Reply:
I want to confirm the ldap api behavior, if the input mail address is invalid (none, wrong mail id etc), what will happen, error will be raised or return none instead. current logic is two steps operation, we need to handle the case that manager id is none but no error raised.

Reply:
the `conn.entries` returns empty list.

Reply:
So it means, no ldap exception will be raised. the manager UID could be None, am I right?

Reply:
Yes, there shouldn't be any exception

Reply:
ok, the error handling logic should be enhanced here, do validation for `manager_id` before calling `_get_user_email `, log warning message and return directly or raise exception. I prefer option#1, if there are multiple unverified cve issues, the process will not exit

Reply:
I fixed the class. Now the process is not interrupted. If a record is not found for specified parameters, `None` is returned.  In the notification class, warning is send:
`logger.warning(f"Manager email was not found for user {qa_contact_email}")`